src/vehicle.cpp

```diff
@@ -3756,9 +3764,35 @@ void vehicle::idle(bool on_map) {
 
     if( on_map ) {
         update_time();
+        if(scoop_on){
+            operate_scoop();
+        }
+    }
+}
+void vehicle::operate_scoop(){
+    std::vector<int> scoops=all_parts_with_feature("SCOOP");
+    auto c=get_points();
+    for(int scoop:scoops){
+        auto position = global_pos3()+parts[scoop].precalc[0];
+        const char *sound_msgs[]={"Whirrrr","Ker-chunk","Swish","Cugugugugug","Horrifying"};
+        sounds::sound(position, (int) rng(20,35), sound_msgs[rng(0,3)]);
+        for(tripoint i:c){
+            if(g->m.moppable_items_at(i)){//That's right, street sweepers can now clean streets
+                g->m.mop_spills(i);
+                continue;//the item should not be accessed at this point.
+            }
+            if( g->m.has_items(i) ){
+                item* that_item_there = g->m.item_from(i,0);//remove the first item on each square.
```

Comment:
Also, lifting multiple small items should work. The scoop lifting 30 skewers should be fine, but lifting one zombie corpse shouldn't.